### ISSUE TITLE
Unable to deploy Packageserver on Kubernetes

### ISSUE THREAD

**Issue by yeplaa** [operator-framework/operator-lifecycle-manager#1668]
## Type of question

Deployment OLM on kubernetes 1.18

## Question

**What did you do?**
I want install OLM with 'customizing OLM installation' method.

**What did you expect to see?**
The packageserver installation by his CSV in a olm namespace.

**What did you see instead? Under which circumstances?**
Currently the catalog-operator and olm-catalog are OK : 

k get pods -n olm
NAME                               READY   STATUS    RESTARTS   AGE
catalog-operator-7476959cd-qrlzx   1/1     Running   0          52m
olm-operator-847b79bd86-vvf2c      1/1     Running   0          27m

But the packageserver csv is failed : 
 k get csv
NAME            DISPLAY          VERSION   REPLACES   PHASE
packageserver   Package Server   1.0.0                Failed

I have this errors : 

```
- lastTransitionTime: "2020-07-22T16:56:34Z"
    lastUpdateTime: "2020-07-22T16:56:34Z"
    message: apiservice.apiregistration.k8s.io "v1.packages.operators.coreos.com"
      not found
    phase: Pending
    reason: APIServiceResourcesNeedReinstall
  - lastTransitionTime: "2020-07-22T16:56:34Z"
    lastUpdateTime: "2020-07-22T16:56:34Z"
    message: all requirements found, attempting install
    phase: InstallReady
    reason: AllRequirementsMet
  - lastTransitionTime: "2020-07-22T16:56:35Z"
    lastUpdateTime: "2020-07-22T16:56:35Z"
    message: 'install strategy failed: cannot get RoleBinding with a NullRoleBindingNamespaceLister'
    phase: Failed
    reason: InstallComponentFailed
  - lastTransitionTime: "2020-07-22T16:56:41Z"
    lastUpdateTime: "2020-07-22T16:56:41Z"
    message: apiservice.apiregistration.k8s.io "v1.packages.operators.coreos.com"
      not found
    phase: Pending
```

I don't understand what NullRoleBindingNamespaceLister means? 

**Environment**
* operator-lifecycle-manager version: 0.15.1

* Kubernetes version information:

```
k version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2-0-g52c56ce", GitCommit:"592b165660fbf31430b78575e311837d6e73514f", GitTreeState:"clean", BuildDate:"2020-07-13T19:06:44Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}. 
```
  

* Kubernetes cluster kind: 

**Additional context**
Bellow the my-values.yaml file used for installing the OLM : 
```

installType: upstream
# sets the apiversion to use for rbac-resources. Change to `authorization.openshift.io` for openshift
rbacApiVersion: rbac.authorization.k8s.io
# namespace is the namespace the operators will _run_
namespace: olm
# watchedNamespaces is a comma-separated list of namespaces the operators will _watch_ for OLM resources.
# Omit to enable OLM in all namespaces
watchedNamespaces: olm,dbms1,dbms2
# catalog_namespace is the namespace where the catalog operator will look for global catalogs.
# entries in global catalogs can be resolved in any watched namespace
catalog_namespace: catalog-operator
# operator_namespace is the namespace where the operator runs
operator_namespace: operator
# Add
imagestream: false
writeStatusName: '""'
writePackageServerStatusName: ""

# OLM operator run configuration
olm:
  # OLM operator doesn't do any leader election (yet), set to 1
  replicaCount: 1
  # The image to run. If not building a local image, use sha256 image references
  image:
    ref: quay.io/operator-framework/olm@sha256:2c389d2e380c842cbf542820ad4493249164302ddf0e699b0a37105d234e67ee
    pullPolicy: IfNotPresent
  service:
    # port for readiness/liveness probes
    internalPort: 8080

# catalog operator run configuration
catalog:
  # Catalog operator doesn't do any leader election (yet), set to 1
  replicaCount: 1
  # The image to run. If not building a local image, use sha256 image references
  image:
    ref: quay.io/operator-framework/olm@sha256:2c389d2e380c842cbf542820ad4493249164302ddf0e699b0a37105d234e67ee
    pullPolicy: IfNotPresent
  service:
    # port for readiness/liveness probes
    internalPort: 8080
package:
  replicaCount: 2
  image:
    ref: quay.io/operator-framework/olm@sha256:2c389d2e380c842cbf542820ad4493249164302ddf0e699b0a37105d234e67ee
    pullPolicy: Always
  service:
    internalPort: 5443
    securityContext:
          runAsUser: 1000
catalog_sources:
- rh-operators
```

**Comment by yeplaa** [comment]
I found my problem. It's because of watchedNamespaces parameter in my my-values.yaml file. If i delete this entry, it works.
Is it normally? Is this parameter is deprecated? 

Thanks

**Comment by timflannagan** [comment]
@yeplaa Yeah, I was able to reproduce this as well using OLM's master branch when I specified a custom watchedNamespace helm value:

```diff
git diff doc
diff --git a/doc/install/local-values.yaml b/doc/install/local-values.yaml
index 7b47f2a94..9ac927af9 100644
--- a/doc/install/local-values.yaml
+++ b/doc/install/local-values.yaml
@@ -2,6 +2,7 @@ installType: upstream
 rbacApiVersion: rbac.authorization.k8s.io
 namespace: olm
 writeStatusName: '""'
+watchedNamespaces: olm,dbms1,dbms2
 catalog_namespace: olm
 operator_namespace: operators
 debug: true
```

And ran `make run-local` to deploy OLM using those helm chart values. It's unclear to me whether this is something we'd like to support, but looking at the OLM Operator codebase, it looks like we're correctly passing each individual namespace when building up informers and listers, but the registration logic either has a bug and the lister is attempting to access an empty field, or something else is going on. I would recommend continuing to deploy OLM using the default watch namespace configuration while we figure out whether this behavior is still supported or not.

cc @njhale who likely knows this space the best.
